x-pack/plugins/security_solution/public/detections/components/alerts_kpis/common/hooks.ts

```diff
@@ -48,3 +53,27 @@ export const useInspectButton = ({
     };
   }, [setQuery, loading, response, request, refetch, uniqueQueryId, deleteQuery]);
 };
+
+export const useStackByFields = () => {
+  const [stackByFieldOptions, setStackByFieldOptions] = useState<
+    undefined | EuiComboBoxOptionOption[]
+  >(undefined);
```

Comment:
I was thinking about a way to get rid of `undefined` here. If this piece of state is initialized later in the function, we can eliminate it, I think. Something similar like this:

```
function fieldsToOptions(fields) {
  return Object.entries(allFields).reduce<EuiComboBoxOptionOption[]>(
    (filteredOptions: EuiComboBoxOptionOption[], [key, field]) => {
      if (field.aggregatable === true) {
        return [...filteredOptions, { label: key, value: key }];
      } else {
        return filteredOptions;
      }
    },
    []
  );
}

export const useStackByFields = () => {
  const { pathname } = useLocation();

  const { browserFields } = useSourcererDataView(getScopeFromPath(pathname));

  const allFields = useMemo(() => getAllFieldsByName(browserFields), [browserFields]);

  const [stackByFieldOptions, setStackByFieldOptions] = useState(() => fieldsToOptions(allFields));

  useEffect(() => {
    setStackByFieldOptions(fieldsToOptions(allfields));
  }, [allFields]);

  return useMemo(() => stackByFieldOptions, [stackByFieldOptions]);
};
```

In case `allFields` does not change, we can even remove the `useEffect` in there. Wdyt?